cdc release-6.2 branch panic

### What did you do?

3 cdc instances, perform upgrade pd.

### What did you expect to see?

works normally

### What did you see instead?

cdc instance crashed since panic.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2b6aed3]

goroutine 61 [running]:
github.com/pingcap/tiflow/cdc/processor.(*processor).Close(0xc00035b2c0)
	github.com/pingcap/tiflow/cdc/processor/processor.go:988 +0x93
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).closeProcessor(0xc000b414c0, {{0xc000747dd2?, 0xc000410b40?}, {0xc000747dea?, 0x0?}})
	github.com/pingcap/tiflow/cdc/processor/manager.go:145 +0xe6
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick(0xc000b414c0, {0x7fca640d1150?, 0xc000c9a1e0}, {0x3c3ac08?, 0xc0004761e0})
	github.com/pingcap/tiflow/cdc/processor/manager.go:118 +0x447
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc000b22000, {0x7fca640d1150, 0xc000c9a1e0}, 0xc000d442d0, 0x5f5e100, {0x34cb8fa, 0x9})
	github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:266 +0xe63
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc00099c000, {0x3c6c4e0?, 0xc000c9a1e0}, {0x7fca641170d8, 0xc000b414c0}, {0x3c3ac08, 0xc0004761e0}, 0x457925?, {0x34cb8fa, 0x9})
	github.com/pingcap/tiflow/cdc/capture/capture.go:535 +0x185
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
	github.com/pingcap/tiflow/cdc/capture/capture.go:367 +0x30e
created by github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run
	github.com/pingcap/tiflow/cdc/capture/capture.go:350 +0x42a
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x2b6aed3]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
release-6.2
```

```
[2022/08/11 22:03:53.863 +08:00] [INFO] [helper.go:49] ["init log"] [file=/root/tidb-deploy/cdc-8300/log/cdc.log] [level=info]
[2022/08/11 22:03:53.863 +08:00] [INFO] [version.go:47] ["Welcome to Change Data Capture (CDC)"] [release-version=v6.2.0] [git-hash=c0eaa0075d46e3011225796c7f060e2a5d903a77] [git-branch=heads/refs/tags/v6.2.0] [utc-build-time="2022-08-08 10:05:10"] [go-version="go version go1.18.5 linux/amd64"] [failpoint-build=false]
[2022/08/11 22:03:53.863 +08:00] [INFO] [server.go:90] ["creating CDC server"] [pd="[http://lingjin-0.jinling.tispace:2379,http://lingjin-1.jinling.tispace:2379]"] [config="{\"addr\":\"0.0.0.0:8300\",\"advertise-addr\":\"lingjin-3.jinling.tispace:8300\",\"log-file\":\"/root/tidb-deploy/cdc-8300/log/cdc.log\",\"log-level\":\"info\",\"log\":{\"file\":{\"max-size\":300,\"max-days\":0,\"max-backups\":0},\"error-output\":\"stderr\"},\"data-dir\":\"/root/tidb-data/cdc-8300\",\"gc-ttl\":86400,\"tz\":\"System\",\"capture-session-ttl\":10,\"owner-flush-interval\":200000000,\"processor-flush-interval\":100000000,\"sorter\":{\"num-concurrent-worker\":4,\"chunk-size-limit\":134217728,\"max-memory-percentage\":30,\"max-memory-consumption\":17179869184,\"num-workerpool-goroutine\":16,\"sort-dir\":\"/tmp/sorter\"},\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"cert-allowed-cn\":null},\"per-table-memory-quota\":10485760,\"kv-client\":{\"worker-concurrent\":8,\"worker-pool-size\":0,\"region-scan-limit\":40,\"region-retry-duration\":60000000000},\"debug\":{\"table-actor\":{\"event-batch-size\":32},\"enable-db-sorter\":true,\"db\":{\"count\":8,\"concurrency\":128,\"max-open-files\":10000,\"block-size\":65536,\"block-cache-size\":4294967296,\"writer-buffer-size\":8388608,\"compression\":\"snappy\",\"target-file-size-base\":8388608,\"write-l0-slowdown-trigger\":2147483647,\"write-l0-pause-trigger\":2147483647,\"compaction-l0-trigger\":160,\"compaction-deletion-threshold\":10485760,\"compaction-period\":1800,\"iterator-max-alive-duration\":10000,\"iterator-slow-read-duration\":256},\"enable-new-scheduler\":true,\"messages\":{\"client-max-batch-interval\":10000000,\"client-max-batch-size\":8388608,\"client-max-batch-count\":128,\"client-retry-rate-limit\":1,\"server-max-pending-message-count\":102400,\"server-ack-interval\":100000000,\"server-worker-pool-size\":4},\"enable-scheduler-v3\":true,\"scheduler\":{\"heartbeat-tick\":2,\"max-task-concurrency\":10,\"check-balance-interval\":60000000000}},\"cluster-id\":\"default\"}"]
[2022/08/11 22:03:53.867 +08:00] [INFO] [server.go:350] ["/root/tidb-data/cdc-8300 is set as data-dir (390GB available), sort-dir=/root/tidb-data/cdc-8300/tmp/sorter. It is recommended that the disk for data-dir at least have 500GB available space"]
[2022/08/11 22:03:53.868 +08:00] [INFO] [server.go:223] ["http server is running"] [addr=0.0.0.0:8300]
[2022/08/11 22:03:53.871 +08:00] [INFO] [upstream.go:115] ["upstream is initializing"] [upstreamID=0]
[2022/08/11 22:03:53.871 +08:00] [INFO] [client.go:392] ["[pd] create pd client with endpoints"] [pd-address="[http://lingjin-0.jinling.tispace:2379,http://lingjin-1.jinling.tispace:2379]"]
[2022/08/11 22:03:53.876 +08:00] [INFO] [base_client.go:350] ["[pd] switch leader"] [new-leader=http://lingjin-1.jinling.tispace:2379] [old-leader=]
[2022/08/11 22:03:53.876 +08:00] [INFO] [base_client.go:105] ["[pd] init cluster id"] [cluster-id=7130590285930132585]
[2022/08/11 22:03:53.876 +08:00] [INFO] [client.go:687] ["[pd] tso dispatcher created"] [dc-location=global]
[2022/08/11 22:03:53.876 +08:00] [INFO] [upstream.go:151] ["upstream's PDClient created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.882 +08:00] [ERROR] [upstream.go:160] ["init upstream error"] [error="[CDC:ErrVersionIncompatible]version is incompatible: PD 6.3.0 is not supported, the maximum compatible version is 6.3.0-alpha"] [errorVerbose="[CDC:ErrVersionIncompatible]version is incompatible: PD 6.3.0 is not supported, the maximum compatible version is 6.3.0-alpha\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/version.CheckPDVersion\n\tgithub.com/pingcap/tiflow/pkg/version/check.go:146\ngithub.com/pingcap/tiflow/pkg/version.CheckClusterVersion\n\tgithub.com/pingcap/tiflow/pkg/version/check.go:87\ngithub.com/pingcap/tiflow/pkg/upstream.initUpstream\n\tgithub.com/pingcap/tiflow/pkg/upstream/upstream.go:156\ngithub.com/pingcap/tiflow/pkg/upstream.(*Manager).AddDefaultUpstream\n\tgithub.com/pingcap/tiflow/pkg/upstream/manager.go:91\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).reset\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:203\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).Run\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:295\ngithub.com/pingcap/tiflow/cdc/server.(*server).run.func1\n\tgithub.com/pingcap/tiflow/cdc/server/server.go:276\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220513210516-0976fa681c29/errgroup/errgroup.go:74\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/08/11 22:03:53.882 +08:00] [INFO] [client.go:392] ["[pd] create pd client with endpoints"] [pd-address="[lingjin-0.jinling.tispace:2379,lingjin-1.jinling.tispace:2379]"]
[2022/08/11 22:03:53.885 +08:00] [INFO] [base_client.go:350] ["[pd] switch leader"] [new-leader=http://lingjin-1.jinling.tispace:2379] [old-leader=]
[2022/08/11 22:03:53.885 +08:00] [INFO] [base_client.go:105] ["[pd] init cluster id"] [cluster-id=7130590285930132585]
[2022/08/11 22:03:53.886 +08:00] [INFO] [client.go:687] ["[pd] tso dispatcher created"] [dc-location=global]
[2022/08/11 22:03:53.887 +08:00] [INFO] [upstream.go:168] ["upstream's KVStorage created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.887 +08:00] [INFO] [upstream.go:171] ["upstream's GrpcPool created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.887 +08:00] [INFO] [upstream.go:174] ["upstream's RegionCache created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.888 +08:00] [INFO] [upstream.go:181] ["upstream's PDClock created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.888 +08:00] [INFO] [upstream.go:184] ["upstream's GCManager created"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.889 +08:00] [INFO] [pd_api_client.go:98] ["Succeed to add meta region label to PD"]
[2022/08/11 22:03:53.889 +08:00] [INFO] [upstream.go:206] ["upstream initialize successfully"] [upstreamID=7130590285930132585]
[2022/08/11 22:03:53.889 +08:00] [INFO] [manager.go:97] ["default upstream is added"] [id=7130590285930132585]
[2022/08/11 22:03:53.903 +08:00] [INFO] [capture.go:269] ["init capture"] [captureID=443fb242-934d-48d7-b91a-33ad933e6fb5] [captureAddr=lingjin-3.jinling.tispace:8300]
[2022/08/11 22:03:53.910 +08:00] [INFO] [reactor_state.go:88] ["remote capture online"] [captureID=443fb242-934d-48d7-b91a-33ad933e6fb5] [info="{\"id\":\"443fb242-934d-48d7-b91a-33ad933e6fb5\",\"address\":\"lingjin-3.jinling.tispace:8300\",\"version\":\"v6.2.0\"}"]
[2022/08/11 22:03:53.910 +08:00] [INFO] [reactor_state.go:88] ["remote capture online"] [captureID=5a56e487-09ab-4516-8262-b0bc01e55a77] [info="{\"id\":\"5a56e487-09ab-4516-8262-b0bc01e55a77\",\"address\":\"lingjin-5.jinling.tispace:8300\",\"version\":\"v6.2.0\"}"]
[2022/08/11 22:03:53.910 +08:00] [INFO] [reactor_state.go:88] ["remote capture online"] [captureID=7797cb61-c588-4145-84ff-6bee3a4219e5] [info="{\"id\":\"7797cb61-c588-4145-84ff-6bee3a4219e5\",\"address\":\"lingjin-4.jinling.tispace:8300\",\"version\":\"v6.2.0\"}"]
[2022/08/11 22:03:53.911 +08:00] [INFO] [reactor_state.go:125] ["new upstream is add"] [upstream=7130590285930132585] [info="{\"id\":7130590285930132585,\"pd-endpoints\":\"http://lingjin-0.jinling.tispace:2379,http://lingjin-1.jinling.tispace:2379\",\"key-path\":\"\",\"cert-path\":\"\",\"ca-path\":\"\",\"cert-allowed-cn\":null}"]
[2022/08/11 22:03:54.011 +08:00] [ERROR] [processor.go:542] ["run processor failed"] [changefeed=blackhole-test] [capture=lingjin-3.jinling.tispace:8300] [error="[CDC:ErrVersionIncompatible]version is incompatible: PD 6.3.0 is not supported, the maximum compatible version is 6.3.0-alpha"] [errorVerbose="[CDC:ErrVersionIncompatible]version is incompatible: PD 6.3.0 is not supported, the maximum compatible version is 6.3.0-alpha\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/version.CheckPDVersion\n\tgithub.com/pingcap/tiflow/pkg/version/check.go:146\ngithub.com/pingcap/tiflow/pkg/version.CheckClusterVersion\n\tgithub.com/pingcap/tiflow/pkg/version/check.go:87\ngithub.com/pingcap/tiflow/pkg/upstream.initUpstream\n\tgithub.com/pingcap/tiflow/pkg/upstream/upstream.go:156\ngithub.com/pingcap/tiflow/pkg/upstream.(*Manager).AddDefaultUpstream\n\tgithub.com/pingcap/tiflow/pkg/upstream/manager.go:91\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).reset\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:203\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).Run\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:295\ngithub.com/pingcap/tiflow/cdc/server.(*server).run.func1\n\tgithub.com/pingcap/tiflow/cdc/server/server.go:276\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220513210516-0976fa681c29/errgroup/errgroup.go:74\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/08/11 22:03:54.011 +08:00] [INFO] [capture.go:618] ["processor manager closed"] [captureID=443fb242-934d-48d7-b91a-33ad933e6fb5]
[2022/08/11 22:03:54.011 +08:00] [INFO] [client.go:246] ["WatchWithChan exited"] [role=processor]
[2022/08/11 22:03:54.012 +08:00] [INFO] [capture.go:624] ["table actor system closed"] [captureID=443fb242-934d-48d7-b91a-33ad933e6fb5]
```

--- a/cdc/capture/capture.go
+++ b/cdc/capture/capture.go
@@ -483,9 +483,9 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
 		if resignErr := c.resign(resignCtx); resignErr != nil {
 			if errors.Cause(resignErr) != context.DeadlineExceeded {
 				log.Info("owner resign failed", zap.String("captureID", c.info.ID),
-					zap.Error(err), zap.Int64("ownerRev", ownerRev))
+					zap.Error(resignErr), zap.Int64("ownerRev", ownerRev))
 				cancel()
-				return errors.Trace(err)
+				return errors.Trace(resignErr)
 			}
 
 			log.Warn("owner resign timeout", zap.String("captureID", c.info.ID),

--- a/cdc/processor/processor.go
+++ b/cdc/processor/processor.go
@@ -475,6 +475,7 @@ func isProcessorIgnorableError(err error) bool {
 // the `state` parameter is sent by the etcd worker, the `state` must be a snapshot of KVs in etcd
 // The main logic of processor is in this function, including the calculation of many kinds of ts, maintain table pipeline, error handling, etc.
 func (p *processor) Tick(ctx cdcContext.Context, state *orchestrator.ChangefeedReactorState) (orchestrator.ReactorState, error) {
+	p.changefeed = state
 	// check upstream error first
 	if err := p.upstream.Error(); err != nil {
 		return p.handleErr(state, err)
@@ -493,7 +494,6 @@ func (p *processor) Tick(ctx cdcContext.Context, state *orchestrator.ChangefeedR
 		return state, nil
 	}
 	startTime := time.Now()
-	p.changefeed = state
 	state.CheckCaptureAlive(ctx.GlobalVars().CaptureInfo.ID)
 	ctx = cdcContext.WithChangefeedVars(ctx, &cdcContext.ChangefeedVars{
 		ID:   state.ID,

--- a/cdc/processor/processor.go
+++ b/cdc/processor/processor.go
@@ -656,7 +656,7 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 		return errors.Trace(err)
 	}
 
-	stdCtx := contextutil.PutChangefeedIDInCtx(ctx, p.changefeed.ID)
+	stdCtx := contextutil.PutChangefeedIDInCtx(ctx, p.changefeedID)
 	stdCtx = contextutil.PutRoleInCtx(stdCtx, util.RoleProcessor)
 
 	p.mounter = entry.NewMounter(p.schemaStorage,
@@ -668,7 +668,7 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 
 	log.Info("processor try new sink",
 		zap.String("namespace", p.changefeedID.Namespace),
-		zap.String("changefeed", p.changefeed.ID.ID))
+		zap.String("changefeed", p.changefeedID.ID))
 
 	start := time.Now()
 	conf := config.GetGlobalServerConfig()
@@ -676,7 +676,7 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 		log.Info("Try to create sinkV1")
 		p.sinkV1, err = sinkv1.New(
 			stdCtx,
-			p.changefeed.ID,
+			p.changefeedID,
 			p.changefeed.Info.SinkURI,
 			p.changefeed.Info.Config,
 			errCh,
@@ -691,7 +691,7 @@ func (p *processor) lazyInitImpl(ctx cdcContext.Context) error {
 	if err != nil {
 		log.Info("processor new sink failed",
 			zap.String("namespace", p.changefeedID.Namespace),
-			zap.String("changefeed", p.changefeed.ID.ID),
+			zap.String("changefeed", p.changefeedID.ID),
 			zap.Duration("duration", time.Since(start)))
 		return errors.Trace(err)
 	}
@@ -1036,7 +1036,7 @@ func (p *processor) refreshMetrics() {
 func (p *processor) Close() error {
 	log.Info("processor closing ...",
 		zap.String("namespace", p.changefeedID.Namespace),
-		zap.String("changefeed", p.changefeed.ID.ID))
+		zap.String("changefeed", p.changefeedID.ID))
 	for _, tbl := range p.tables {
 		tbl.Cancel()
 	}

--- a/pkg/upstream/upstream.go
+++ b/pkg/upstream/upstream.go
@@ -158,6 +158,7 @@ func initUpstream(ctx context.Context, up *Upstream, gcServiceID string) error {
 	if err != nil {
 		up.err.Store(err)
 		log.Error("init upstream error", zap.Error(err))
+		return errors.Trace(err)
 	}
 
 	up.KVStorage, err = kv.CreateTiStore(strings.Join(up.PdEndpoints, ","), up.SecurityConfig)

